tools/cli/commands/build.js

```diff
@@ -768,13 +768,39 @@ async function buildProject( t ) {
 		if ( composerJson.repositories.length === 0 ) {
 			delete composerJson.repositories;
 		}
-		await writeFileAtomic(
-			`${ buildDir }/composer.json`,
-			JSON.stringify( composerJson, null, '\t' ) + '\n',
-			{ encoding: 'utf8' }
-		);
 	}
 
+	// Update dependency version numbers in composer.json.
+	const composerDepTyes = [ 'require', 'require-dev' ];
+	for ( const key of composerDepTyes ) {
+		if ( composerJson[ key ] ) {
+			for ( const [ pkg ] of Object.entries( composerJson[ key ] ) ) {
+				Object.values( t.ctx.versions ).find( ctxPkg => {
+					if ( ctxPkg.name === pkg ) {
+						let massagedVer = ctxPkg.version;
+
+						// Truncate non-0.x versions to be two components only.
```

Comment:
This makes some sense for PHP packages, but probably not for plugins and other projects.

Maybe not even for require-dev in PHP packages.

Reply:
You mean the version truncation? And if so what would be a way to differentiate the project types?

Reply:
Gave this a shot in d25066f57e37ce9c98cf0a17b737e640baace349